TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/MovementService.java

```diff
@@ -0,0 +1,194 @@
+package com.novoda.tpbot.bot;
+
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.hardware.usb.UsbDevice;
+import android.hardware.usb.UsbDeviceConnection;
+import android.hardware.usb.UsbManager;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+
+import com.felhr.usbserial.UsbSerialDevice;
+import com.felhr.usbserial.UsbSerialInterface;
+import com.novoda.notils.logger.simple.Log;
+import com.novoda.notils.logger.toast.Toaster;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MovementService extends Service {
+
+    private static final String ACTION_USB_PERMISSION = "com.novoda.tpbot.USB_PERMISSION";
+    private static final List<Integer> SUPPORTED_VENDOR_IDS = Arrays.asList(9025, 10755, 4292); // TODO: read from devices_filter.xml
+
+    private UsbDevice device;
+    private UsbDeviceConnection connection;
+    private UsbManager usbManager;
+    private UsbSerialDevice serialPort;
+    private boolean isSerialStarted;
+    private PendingIntent pendingIntent;
+
+    private Toaster toaster;
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        toaster = Toaster.newInstance(MovementService.this);
+        toaster.popToast(MovementService.class.getSimpleName() + " started");
+
+        pendingIntent = PendingIntent.getBroadcast(this, 0, new Intent(ACTION_USB_PERMISSION), 0);
+        IntentFilter filter = new IntentFilter(ACTION_USB_PERMISSION);
+        filter.addAction(UsbManager.ACTION_USB_DEVICE_ATTACHED);
+        filter.addAction(UsbManager.ACTION_USB_DEVICE_DETACHED);
+        registerReceiver(broadcastReceiver, filter);
+
+        startConnection();
+
+        return START_STICKY;
+    }
+
+    public void startConnection() {
+
+        if (!isSerialStarted) {
+            usbManager = (UsbManager) getSystemService(Context.USB_SERVICE);
+
+            HashMap<String, UsbDevice> usbDevices = usbManager.getDeviceList();
+            if (!usbDevices.isEmpty()) {
```

Comment:
I would also have a quick return above too. Always try to prefer the positive case when possible 😃